internal/pkg/argocd/argocd.go

```diff
@@ -170,26 +172,90 @@ func generateDiffOfAComponent(ctx context.Context, componentPath string, prBranc
 	}
 	foundApps, err := appIf.List(ctx, &appLabelQuery)
 	if err != nil {
-		log.Errorf("Error listing ArgoCD applications: %v", err)
-		componentDiffResult.DiffError = err
-		return componentDiffResult
+		return nil, fmt.Errorf("Error listing ArgoCD applications: %v", err)
+
 	}
 	if len(foundApps.Items) == 0 {
-		componentDiffResult.DiffError = fmt.Errorf("No ArgoCD application found for component path %s(repo %s), used this label selector: %s", componentPath, repo, labelSelector)
+		return nil, fmt.Errorf("No ArgoCD application found for component path sha1 %s(repo %s), used this label selector: %s", componentPathSha1, repo, labelSelector)
+	}
+
+	// we expect only one app with this label and repo selectors
+	return &foundApps.Items[0], nil
+}
+
+// This is the default method to find an ArgoCD application by the manifest-generate-paths annotation.
+// It assume the ArgoCD (optional) manifest-generate-paths annotation is set on all relevant apps.
+// Notice that this method include a full list all ArgoCD applications in the repo, this could be a performance issue if there are many apps in the repo.
```

Comment:
following [godoc convention](https://github.com/commercetools/special-delivery/blob/main/standards/CODE.md#code-comments-should-follow-godoc-conventions) for functions ([example](https://github.com/efficientgo/examples/blob/6a9ad11522ce13e274fa6557b460a83a5a4cb429/pkg/godoc/block.go#L29)): 
```suggestion
// findArgocdAppByManifestPathAnnotation is the default method to find an ArgoCD application by the manifest-generate-paths annotation.
// It assumes the ArgoCD (optional) manifest-generate-paths annotation is set on all relevant apps.
// Notice that this method includes a full list of all ArgoCD applications in the repo, this could be a performance issue if there are many apps in the repo.
```